resolwe_bio/processes/samtools/samtools_mpileup.py

```diff
@@ -0,0 +1,367 @@
+                default=0,
+            )
+            output_mapping_qual = BooleanField(
+                label="Output mapping quality",
+                description="Output mapping quality as a separate column in the pileup output. [-s]",
+                default=False,
+            )
+            output_read_names = BooleanField(
+                label="Output read names",
+                description="Output read names in the pileup output. [--output-QNAME]",
+                default=False,
+            )
+            output_zero_depth = BooleanField(
+                label="Output all positions (including those with zero depth)",
+                description="Output all positions including those with zero depth. [-a]",
+                default=False,
+                disabled="advanced.output_all_positions == true",
+            )
+            output_all_positions = BooleanField(
+                label="Output absolutely all positions",
+                description="Output absolutely all positions, including unused reference sequences. "
+                "Note that when used in conjunction with a BED file the Output all positions option may "
+                "sometimes operate as if Output absolutely all positions was specified. This can happen "
+                "if the reference sequence has coverage outside of the region specified in the BED file. [-aa]",
+                default=False,
+            )
+            min_base_qual = IntegerField(
+                label="Minimum base quality for a base to be considered",
+                description="Only count reads with base quality greater than or equal to value specific. [-Q]",
+                required=False,
+            )
+            min_mapping_qual = IntegerField(
+                label="Minimum mapping quality for an alignment to be used",
+                description="Only count reads with mapping quality greater than or equal to value specific. [-q]",
+                required=False,
+            )
+            excl_flags = ListField(
+                StringField(),
+                label="Filter flags",
+                default=["UNMAP", "SECONDARY", "QCFAIL", "DUP"],
+                description="Filter flags: skip reads with mask bits set. "
+                "Press ENTER after each flag. [-ff]",
+            )
+            excl_output_ends = BooleanField(
+                label="Exclude output ends in sequence column",
+                description="Removes the “^” (with mapping quality) and “$” markup from the sequence column [--no-output-ends].",
+                default=False,
+            )
+            output_base_counts = BooleanField(
+                label="Output per base counts",
+                description="Output base counts at each position. "
+                "This is a custom implementation and not part of the original samtools mpileup. "
+                "It outputs a file with per base counts at each position.",
+                default=False,
```

Comment:
Should we make it a `default=True`?

Reply:
As this is a custom implementation, I thought it would be more appropriate to make it false.

Reply:
This is additional functionality, that is not changing the tool main output. I see no harm in enabling this by default. In this regard it is more problematic that the field `excl_flags` adds to the default filtering rules by including `UNMAP`. Perhaps its best to explicitly state this in tool help section.

Reply:
Will remove UMAP from defaults.

Reply:
It is enabled now by default.